File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int solution(int n) 
    {
        List<int> ans = new List<int>();
        while(n / 3 != 0)
        {
            ans.Add(n % 3);
            n /= 3;
        }
        ans.Add(n % 3);
      
        ans.Reverse();
        
        int result = 0;
        int pow = 1;
        for(int i=0; i<ans.Count; i++)
        {
            result += pow * ans[i];
            pow *= 3;
        }
        
        return result;
    }
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n) 
{
    vector<int> vec;
    while(n > 0)
    {
        vec.push_back(n%3);
        n /= 3;
    }
    
    for(int &i : vec)
    {
        cout << i << " ";
    }
    
    int ans = 0;
    int a = 1;
    for(int i=vec.size()-1; i>=0; i--)
    {
        ans += vec[i] * a;
        a *= 3; 
    }
    return ans;
}